tests/unit/visualise/test_dataset_renderer.py

```diff
@@ -3,101 +3,195 @@
-# import geopandas
-# import xarray
+import pandas
+import iris
+from iris.cube import Cube, CubeList
+from shapely.geometry import Polygon, MultiPolygon
 
 import clean_air.visualise.dataset_renderer as dr
+from clean_air.data import DataSubset
+
+
+@pytest.fixture()
+def timeseries_filepath(sampledir):
+    timeseries_filepath = os.path.join(sampledir, "model_full", "aqum_hourly_o3_20200520.nc")
+    return timeseries_filepath
+
+
+@pytest.fixture()
+def diurnal_filepath(sampledir):
+    diurnal_filepath = os.path.join(sampledir, "model_full", "aqum_hourly_o3_48_hours.nc")
+    return diurnal_filepath
+
+
+@pytest.fixture()
+def clean_data(timeseries_filepath):
+    # Note: This is a DataSubset object which can be used and adapted for later fixtures and tests.
+    # These objects are NOT subscriptable.
+    clean_df = DataSubset(timeseries_filepath)
+    return clean_df
+
+
+@pytest.fixture()
+def multiday_data(diurnal_filepath):
+    # Note: This is a DataSubset object which can be used and adapted for later fixtures and tests.
+    # These objects are NOT subscriptable.
+    multiday_df = DataSubset(diurnal_filepath)
+    return multiday_df
+
+
+@pytest.fixture()
+def tmp_output_path(tmp_path):
+    tmp_output_path = tmp_path / "tmp_output_path"
+    tmp_output_path.mkdir()
+    return tmp_output_path
 
 
 class TestDatasetRenderer:
     """
-    Class to test object initialisation in Renderer
+    Class to test object initialisation in Renderer.  This process must be able to identify incoming objects, convert
+    them to lasy iris cubes and extract the coordinate metadata.
     """
 
-    @pytest.fixture(scope="class")
-    def renderer(self, sampledir):
-        path = os.path.join(
-            sampledir,
-            "model_full",
-            "aqum_daily_daqi_mean_20200520.nc"
-        )
-        return dr.Renderer(path)
-
-    def test_lazy_iris_data(self, renderer):
-        # Check that the iris dataset is loaded as lazy data for performance:
-        assert renderer.dataset.has_lazy_data
-
-    def test_found_dim_coords(self, renderer):
-        # Check that all of the iris-guessed coords are the ones that we
-        # expect it to discover:
-        assert renderer.x_coord == 'projection_x_coordinate'
-        assert renderer.y_coord == 'projection_y_coordinate'
+    @pytest.yield_fixture(autouse=True)
+    def setup_class(self, sampledir):
+        """SET UP all inputs to tests"""
+        path = os.path.join(sampledir, "model_full", "aqum_daily_daqi_mean_20200520.nc")
+        self.renderer = dr.Renderer(path)
+
+    def test_lazy_iris_data(self):
+        """GIVEN a dataset with multiple coordinates,
+        WHEN the dataset is loaded into an iris cube to identify coordinates,
+        THEN the iris object is lazily loaded (i.e. metadata loaded without data until needed)."""
+        assert self.renderer.plot_list[0].has_lazy_data
+
+    def test_found_dim_coords(self):
+        """GIVEN a dataset with multiple coordinates,
```

Comment:
Extremely minor point but existing tests with this format have the quotes on a separate line, as in:
"""
GIVEN...

 rather than:

"""GIVEN...